sdk/monitor/azure-monitor-opentelemetry-exporter/src/main/java/com/azure/monitor/opentelemetry/exporter/AzureMonitorTraceExporter.java

```diff
@@ -262,11 +265,17 @@ private void applySemanticConventions(SpanData span, RemoteDependencyData remote
             return;
         }
         String messagingSystem = attributes.get(AttributeKey.stringKey("messaging.system"));
-        // TODO (trask) ideally EventHubs SDK should emit messaging.system attribute
-        if (messagingSystem != null || span.getName().equals("EventHubs.message")) {
+        if (messagingSystem != null) {
             applyMessagingClientSpan(attributes, remoteDependencyData, messagingSystem, span.getKind());
             return;
         }
+        // TODO (trask) ideally EventHubs SDK should conform and fit the above path used for other messaging systems
+        //  but no rush as messaging semantic conventions may still change
+        String name = span.getName();
+        if (name.equals("EventHubs.send") || name.equals("EventHubs.message")) {
+            applyEventHubsSpan(attributes, remoteDependencyData);
+            return;
+        }
```

Comment:
A similar change should be made for Service Bus too.

I have created a couple of issues to track this for Event Hubs and Service Bus. 
https://github.com/Azure/azure-sdk-for-java/issues/21684
https://github.com/Azure/azure-sdk-for-java/issues/21686